src/poprox_recommender/default.py

```diff
@@ -52,17 +54,27 @@ def parse_row(token_mapping, row):
     )
 
 
+# check the input type of articles 
+def is_list_of_dicts(articles): 
+    if isinstance(articles, list) and all(isinstance(item, dict) for item in articles):
+        return True 
+    return False
+
 # prepare the news.tsv with information for NRMS model
 def transform_article_features(articles: List[Article], token_mapping):
-    article_df = pd.DataFrame([article.model_dump() for article in articles])
+    if is_list_of_dicts: 
+        article_df = pd.DataFrame(articles)
+    else:
+        article_df = pd.DataFrame([article.model_dump() for article in articles])
```

Comment:
I'd rather update the code in the test case to create `Article` and `ClickHistory` objects, which will make the input type supplied to this function consistent and avoid needing to check types here. Otherwise we're going to find ourselves changing `click_history.article_ids` to `click_history["article_id"]` and back again every time we want to run the test case and deploy the code.

Reply:
I agree with you. We did not realize that the serverless examples are in JSON format. We will update the test case to use JSON format and modify the relevant parts of the code accordingly. This will ensure consistency in the input types supplied to the function and eliminate the need to switch between click_history.article_ids and click_history["article_id"] each time we run the test case and deploy the code.